### ISSUE TITLE
Gentle background updates

### ISSUE THREAD

**Issue by hasali19** [ImranR98/Obtainium#1105]
Currently when an app is updated in the background, the app will be killed if it is currently running. This isn't great when it closes while I'm in the middle of using it.

Android 14 introduces a [new API](https://developer.android.com/about/versions/14/features#update-apps-timing) to allow 3rd party app stores to defer package installation until the system decides that it won't be disruptive to the user. Obtainium should use this API to make background updates less disruptive.

**Comment by Couchy** [comment]
Yes, please. The new requestUserPreapproval() method will also allow installing even when obtanium itself is in the background.
